### PR TITLE
Upgrade to OCaml 5.1

### PR DESCRIPTION
**Pull request by nawarajshah**
Upgrade to OCaml 5.1

Issue: https://github.com/NathanReb/ocaml-api-watch/issues/28

### CODE REVIEW

**Review by NathanReb — CHANGES_REQUESTED** [review]

File: dune-project [hunk 1]
```diff
@@ -15,6 +15,6 @@
  (name api-watch)
  (synopsis "Libraries and tools to keep watch on your OCaml lib's API changes")
  (depends
-  (ocaml (and (>= 4.14.0) (< 5.0.0)))
+  (ocaml (and (>= 4.14.0) (< 5.2.0)))
```
Comment (NathanReb) [inline dune-project:18]: This is still incorrect, you also need to raise the lower bound from `4.14.0` to `5.1.0`. Here you are allowing all of `4.14`, `5.0` and `5.1`.

**Review by NathanReb — CHANGES_REQUESTED** [review]

File: dune-project (hunk 1 above)
Comment (nawarajshah) [reply dune-project:18]: Sorry to miread your instruction

**Comment by NathanReb** [comment]
We just merged #30 so it seems you will have to rebase your work on top of main before we can merge!

Please let me know if you need help with this process!

**Comment by nawarajshah** [comment]
> We just merged #30 so it seems you will have to rebase your work on top of main before we can merge!
> 
> Please let me know if you need help with this process!

Even though i did Rebase from my side, can you conform it?

**Comment by nawarajshah** [comment]
> We just merged #30 so it seems you will have to rebase your work on top of main before we can merge!
> 
> Please let me know if you need help with this process!

Even though i did Rebase from my side, can you conform it?

**Comment by NathanReb** [comment]
It seems you didn't push your changes since github mentions there are still conflicts!

**Comment by nawarajshah** [comment]
> It seems you didn't push your changes since github mentions there are still conflicts!

can you tell me how to do that?

**Comment by NathanReb** [comment]
@nawarajshah you still haven't pushed your changes.

A rebase is roughly a rewriting of your local git history on your working branch. If you want this to impact this PR you will have to push this change to you remote branch on your fork of the project.

Since the history has been rewritten after your previously pushed, you'll have to run `git push --force` as the remote branch isn't an ancestor of your local branch anymore (i.e. you did not simply add new commits on top of what is on the remote branch).

What do you get when running `git status` in your local copy of the project?

**Comment by nawarajshah** [comment]
> @nawarajshah you still haven't pushed your changes.
> 
> A rebase is roughly a rewriting of your local git history on your working branch. If you want this to impact this PR you will have to push this change to you remote branch on your fork of the project.
> 
> Since the history has been rewritten after your previously pushed, you'll have to run `git push --force` as the remote branch isn't an ancestor of your local branch anymore (i.e. you did not simply add new commits on top of what is on the remote branch).
> 
> What do you get when running `git status` in your local copy of the project?

this Is what i get when i run `git status`
image

**Comment by shonfeder** [comment]
Can you show the output of these commands?

- `git remote -v`
- `git log -1 upstream/main`

**Comment by nawarajshah** [comment]
> Can you show the output of these commands?
> 
> * `git remote -v`
> * `git log -1 upstream/main`

This is the output.

image

**Comment by NathanReb** [comment]
What rebase command did you run? If you rebased on top of your local `main` branch, it's possible that it was simply out of sync with the upstream `main` branch.

You should either update your local `main` branch and rebase again:
```
git checkout main
git pull
git checkout nawaraj_shah
git rebase main
```

or simply rebase on top of the upstream `main` branch:
```
git fetch --all
git rebase origin/main
```

**Comment by nawarajshah** [comment]
> git rebase origin/main

here it is
image

**Comment by nawarajshah** [comment]
> Can you show the output of these commands?
> 
> * `git remote -v`
> * `git log -1 upstream/main`

after I get help from Bard, I can get here

image

**Comment by shonfeder** [comment]
That's progress! This shows what I suspected: that you did not have a remote configured to track the "upstream" repository, and as a result where not able to pull the latest updates. This documentation can help with understanding how to manage remotes and you may find it useful in the future :) https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes

Now, to move forward, from the branch associated with this PR, you should be able to

```
git rebase upstream/main
```

And then finally

```
git push --force
```

Should update this pull request your changes.

**Comment by shonfeder** [comment]
Fixed in an alternate PR.

**Pull request closed without merging**